pkg/service/user/sshkey.go

```diff
@@ -0,0 +1,49 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package user
+
+import (
+	"context"
+
+	"github.com/cockroachdb/errors"
+	"github.com/jackc/pgconn"
+
+	"github.com/tensorchord/envd-server/errdefs"
+	"github.com/tensorchord/envd-server/pkg/query"
+)
+
+func (u *generalService) ListPubKeys(ctx context.Context, loginName string) ([]query.Key, error) {
+	return u.querier.ListKeys(ctx, loginName)
+}
+
+func (u *generalService) CreatePubKey(ctx context.Context, loginName, name string, pubKey []byte) error {
+	_, err := u.querier.CreateKey(ctx, query.CreateKeyParams{
+		LoginName: loginName,
+		PublicKey: pubKey,
+	})
+	if err != nil {
+		var pgErr *pgconn.PgError
+		if errors.As(err, &pgErr) {
+			switch pgErr.Code {
+			case "23505":
```

Comment:
What's this magic number?

Reply:
They have an error code lib https://github.com/jackc/pgerrcode/blob/master/errcode.go

Reply:
OK let's use it.